packages/material-ui/src/Badge/Badge.js

```diff
@@ -371,7 +371,7 @@ Badge.propTypes = {
   overlap: chainPropTypes(
     PropTypes.oneOf(['circle', 'rectangle', 'circular', 'rectangular']),
     (props) => {
-      const { overlap = 'rectangle' } = props;
+      const { overlap = 'rectangular' } = props;
```

Comment:
Isn't this the only change required? I would leave the rest of the code as it was. Also, note that this is the only change that would be included in a new version of v4. We basically now only accept critical fixes on v4. We can however merge it and include it if a new version is going to be created. Let's first ensure which changes are really required.